### ISSUE TITLE
only allow apply on specific branches

### ISSUE THREAD

**Issue by jwermuth** [jwermuth/gradle-plugin-terraform#9]
Use case. 
Team is working on terraform controlled infrastructure. Several new, non trivial, things being developed in parallel. Each new thing is developed in its own branch, e.g. feature-a, feature-b etc. Each new feature, e.g. feature-a merges the main branch into their feature branch regularly, and checks what feature-a would do with plan.
Now something important happens in production (aka the main branch). feature-a team can check out main branch and work on it without the half written stuff on feature-a getting in the way.

Its easy, and can be fatal, to do an apply on the wrong branch in such a case, so it should be possible to "lock" apply to one branch.

Implementation could be done by setting an attribute on the tfApply task, e.g.
```
tfApply() {
 applyBranch='production'
}
and let applyBranch be 'master' by default.

another implementation approach could be to create a configuration closure, e.g.
```
terraform {
  applyBranch='master'
}

**Comment by jwermuth** [comment]
@mbrouer any comments on this

**Comment by mbrouer** [comment]
Sounds good.
Keep in mind that it should always be possible to apply from master/production even if there are no changes since last apply.

**Comment by jwermuth** [comment]
Agree, but I think the issue you mention is its own issue. Please make a new issue on this, so we can discuss how it should behave there.
